book/src/lint_configuration.md

```diff
@@ -676,6 +676,16 @@ The named groupings of different source item kinds within modules.
 * [`arbitrary_source_item_ordering`](https://rust-lang.github.io/rust-clippy/master/index.html#arbitrary_source_item_ordering)
 
 
+## `module-items-ordered-within-groups`
```

Comment:
I was mostly thinking that the config is getting convoluted in either case, but I fully agree that it would make sense to go for higher granularity here. I would like to have an option for "all", but would need to investigate how to best express that in TOML.

Preferrably I'd have the options of:
```TOML
module-items-ordered-within-groups = "none" (being the default)
module-items-ordered-within-groups = ["mod", "use", "my_custom_group", ...]
module-items-ordered-within-groups = "all"
```

I'll put some thought into this in the coming days.